client/src/components/CatsList/CatList.tsx

```diff
@@ -18,9 +18,20 @@ export const CatList = () => {
   const [filterData, setFilterData] = useState<TCat[]>([]);
   const [loading, setLoading] = useState<Boolean>(true);
 
-  const getCats = useCallback(async (): Promise<void> => {
+  const debouncedGetCats = useCallback(
+    debounce((filter) => {
+      getCats(filter);
+    }, 1000),
+    []
+  );
+
+  const handleFilter = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    debouncedGetCats(e.target.value);
+  }, []);
```

Comment:
1, 1000 ms is a lot for debounce
2. debounce is not enough you need to cancel old requests